QueryBuilder/Helper.cs

```diff
@@ -83,57 +62,40 @@ public static bool IsSubclassOfRawGeneric(Type generic, Type toCheck)
                 var cur = IsGenericType(toCheck) ? toCheck.GetGenericTypeDefinition() : toCheck;
 
                 if (generic == cur)
-                {
                     return true;
-                }
 
                 toCheck = BaseType(toCheck);
             }
 
             return false;
         }
 
-        public static List<int> AllIndexesOf(string str, string value)
+        public static IEnumerable<int> AllIndexesOf(string str, string value)
         {
-            if (String.IsNullOrEmpty(value))
-            {
-                return new List<int>();
-            }
+            if (string.IsNullOrEmpty(value))
+                yield break;
 
-            List<int> indexes = new List<int>();
-            for (int index = 0; ; index += value.Length)
+            var index = 0;
+            do
             {
-                index = str.IndexOf(value, index);
+                index = str.IndexOf(value, index, StringComparison.Ordinal);
+
                 if (index == -1)
-                    return indexes;
-                indexes.Add(index);
-            }
+                    yield break;
+
+                yield return index;
+            } while ((index += value.Length) < str.Length);
         }
 
         public static string ReplaceAll(string subject, string match, Func<int, string> callback)
         {
-            if (string.IsNullOrWhiteSpace(subject))
-            {
+            if (string.IsNullOrWhiteSpace(subject) || !subject.Contains(match))
                 return subject;
-            }
-
-            var tokens = subject.Split(new[] { match }, StringSplitOptions.None).ToList();
-
-            if (tokens.Count == 1)
-            {
-                return tokens[0];
-            }
-
-            var newStr = new List<string>();
-            newStr.Add(tokens[0]);
-
-            for (var i = 1; i < tokens.Count; i++)
-            {
-                var replacement = callback.Invoke(i - 1);
-                newStr.Add(replacement + tokens[i]);
-            }
 
-            return string.Join("", newStr);
+            var index = 0;
+            return subject
```

Comment:
feeling that statements like this are harder to understand 🙄

Reply:
Ok, I will refactor with `Select`, `StringBuilder`